### PR TITLE
Bug 1880309: Bump k8s.io/client-go to v0.18.6 to fix bug in reflector not recovering from "Too large resource version"

### PR DESCRIPTION
**Pull request by jsafrane**
To fix recovery from "Too large resource version" error.

Full rebase of client-go has been suggested in https://bugzilla.redhat.com/show_bug.cgi?id=1880309.
Basically backport of https://github.com/openshift/cluster-csi-snapshot-controller-operator/pull/43 to get client-go 0.18.3 + bump client-go to 0.18.6

### CODE REVIEW

**Comment by jsafrane** [comment]
cc @openshift/storage

File: pkg/dependencymagnet/dependencymagnet.go [hunk 1]
```diff
@@ -5,6 +5,7 @@
 package dependencymagnet
 
 import (
-	_ "github.com/jteeuwen/go-bindata/go-bindata"
-	_ "github.com/openshift/library-go/alpha-build-machinery"
+	_ "github.com/go-bindata/go-bindata"
+	_ "github.com/go-bindata/go-bindata/go-bindata"
```
Comment (bertinatto) [inline pkg/dependencymagnet/dependencymagnet.go:9]: bindata twice?

Comment (jsafrane) [reply pkg/dependencymagnet/dependencymagnet.go:8]: nice catch, fixed

**Review by bertinatto — COMMENTED** [review]
/lgtm

**Comment by markmc** [comment]
(patch manager) looks like a genuine user-impacting issue. Not ideal (in terms of risk) that a +88 −29 upstream fix requires such a huge patch to backport, but it seems like this is our best option.

**Comment by markmc** [comment]
/retitle Bug 1880309: Bump k8s.io/client-go to v0.18.6 to fix bug in reflector not recovering from "Too large resource version"

**Pull request merged**